react/ProductWrapper.js

```diff
@@ -48,7 +49,11 @@ const ProductWrapper = ({
     <WrapperContainer className="vtex-product-context-provider">
       <ProductContextProvider query={query} product={product}>
         <Content loading={loading} childrenProps={childrenProps}>
-          {children}
+          <LoadingContextProvider
```

Comment:
this way you are creating a new object at each render, you should useMemo to only create a new context object.